clarity_ext/extensions.py

```diff
@@ -272,20 +275,29 @@ def _run(self, config, path, pid, module, artifacts_to_stdout, disable_context_c
             context.logger.log("Finished running script '{}'".format(module))
             context.commit()
         except UsageError as e:
-            # Commit in order to upload step log
-            # Ordinary files and objects shouldn't have been moved to upload queue,
-            # since the scipt have been interrupted by the usage error.
-            context.commit()
-            # UsageErrors are deferred and handled by the notify method
-            # To support the case (legacy) if someone raises an error without adding it to the defer list,
-            # we add it to the instance too:
-            if len(instance.errors) == 0:
-                instance.errors[e] = list()
+            self._commit_step_log(context, instance, e)
+        except Exception as e:
```

Comment:
If we grab exceptions at the lowest level I think we should always rethrow them. Otherwise we can have some very hard to debug issues later on.